### PR TITLE
Updated Devouring Plague for spriest

### PR DESCRIPTION
**Pull request by dbyena**
Updated devouring plague spell. This will not be the final update of the spell, but I made a note for what needs to be changed.

### CODE REVIEW

**Comment by dbyena** [comment]
I also need to figure out a way to handle the BonusSpellHitRating misery gives when the debuff is active. This will need to be for all spells when misery is active.

File: sim/priest/devouring_plague.go [hunk 1]
```diff
@@ -24,43 +36,45 @@ func (priest *Priest) registerDevouringPlagueSpell() {
...
 				core.TernaryFloat64(priest.Talents.Shadowform, 1.15, 1),
 			ThreatMultiplier: 1 - 0.08*float64(priest.Talents.ShadowAffinity),
 			IsPeriodic:       true,
-			BaseDamage:       core.BaseDamageConfigMagicNoRoll(1216/8, 0.1),
-			OutcomeApplier:   priest.OutcomeFuncTick(),
+			BaseDamage:       core.BaseDamageConfigMagicNoRoll(1376/8, 0.1849),
+
+			// OutcomeApplier: priest.DPcrits()
+			// Assume shadow form is always active for right now
+			//if priest.Talents.Shadowform > 0 { // NEED TO ADD A CHECK TO SEE IF SHADOWFORM TALENT IS CHOSEN, THEN DETERMINE IF DOTS CAN CRIT OR NOT
```
Comment (jimmyt857) [inline sim/priest/devouring_plague.go:65]: The way you can do this is to create a variable outside this definition and then use it here. E.g:

```
outcomeApplier := priest.OutcomeFuncTick()
if priest.Talents.Shadowform {
  outcomeApplier = priest.OutcomeFuncMagicCrit(priest.SpellCritMultiplier(1, 1))
}

...

priest.DevouringPlagueDot = ...
  ...
  OutcomeApplier: outcomeApplier,
  ...
}
```

**Comment by dbyena** [comment]
Need to also increase critical strike chance of the dots with mindmelt talent.

File: sim/priest/devouring_plague.go [hunk 2]
```diff
@@ -47,31 +52,24 @@ func (priest *Priest) registerDevouringPlagueSpell() {
 			ActionID: actionID,
 		}),
 
-		NumberOfTicks: 8,
-		TickLength:    time.Second * priest.ApplyCastSpeed(3),
-
+		NumberOfTicks:       8,
+		TickLength:          time.Second * priest.ApplyCastSpeed(3),
```
Comment (lologarithm) [inline sim/priest/devouring_plague.go:56]: i think you can use the `AffectedByCastSpeed: true` and then set the tick length to just `time.Second*3`

**Pull request merged**